applications/processor/session-filter/src/main/java/com/csgroup/s2pdgs/sessionfilter/SessionFilterProcessor.java

```diff
@@ -1,105 +1,253 @@
+import com.csgroup.s2pdgs.station.SessionInformationBlock;
+
 import org.springframework.messaging.Message;
 import org.springframework.messaging.support.MessageBuilder;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Collections;
 import java.util.List;
 import java.util.function.Function;
 
+import javax.xml.bind.JAXBException;
+
 @Slf4j
-public class SessionFilterProcessor implements Function<Message<SessionData>, List<Message<SessionData>>> {
+public class SessionFilterProcessor implements Function<Message<StationProduct>, List<Message<SessionData>>> {
+    private static final int NB_DSIB_PER_SESSION = 2;
 
     private final Function<String, Catalog> catalogById;
     private final AssetCatalogClientManager catalogClientManager;
-    private final Cache<String, SessionData> cache;
 
     public SessionFilterProcessor(
         Function<String, Catalog> catalogById,
-        AssetCatalogClientManager catalogClientManager,
-        Cache<String, SessionData> cache
+        AssetCatalogClientManager catalogClientManager
     ) {
         this.catalogById = catalogById;
         this.catalogClientManager = catalogClientManager;
-        this.cache = cache;
     }
 
     @Override
-    public List<Message<SessionData>> apply(Message<SessionData> msg) {
-        var session = msg.getPayload();
-        log.debug("Filter found session {}", session);
-        var catalog = catalogById.apply(session.getCatalogId());
-        if(cache.getIfPresent(session.getSessionId()) != null){
-            log.debug("Duplicated session {}", session.getSessionId());
-            return Collections.emptyList();
-        }
-        else {
-            cache.put(session.getSessionId(), session);
-            log.debug("Filter found new session {}", session.getSessionId());
-
-            getSessionAsset(catalog, session)
-                .map(StationProduct::getSession)
-                .filter(AcquisitionSession::isDownloaded)
-                .peek(acqSession -> {
-                    log.debug("Removing raws from catalog for session {}", acqSession);
-                    removeRawsFromCatalog(catalog, session);
-                });
-
-            return List.of(MessageBuilder.withPayload(session).build());
-        }
+    public List<Message<SessionData>> apply(Message<StationProduct> msg) {
+        var product = msg.getPayload();
+        log.debug("Filter found product {}", product);
+
+        var catalogId = product.getCatalogId();
+        var catalog = catalogById.apply(catalogId);
+        
+        // First check if session was done before (done == in elastic)
+        // Otherwise check session completeness
+        return getSessionAsset(catalog, product)
+                // If found in elastic == session already done
+                .map(asset -> createEmptyList())
```

Comment:
Avec la `List` de Java on peut faire 

```suggestion
                .map(asset ->List.<Message<SessionData>>of())
```